files/02_amend_replication_agreements_and_reinitialize.sh

```diff
@@ -0,0 +1,119 @@
+      # Extract the part where the fractional replication attributes
+      # are specified
+      old_repl_attr=$(sed --quiet \
+        "s/^[[:blank:]]*nsDS5ReplicatedAttributeList:[[:blank:]]*\(.*\)$/\1/p" \
+        <<< "$cmd_output")
+      # Remove krblastsuccessfulauth from the list of excluded
+      # fractional replication attributes
+      new_repl_attr=${old_repl_attr// krblastsuccessfulauth/}
+      # Extract the part where the total replication attributes are
+      # specified
+      old_repl_attr_total=$(sed --quiet \
+        "s/^[[:blank:]]*nsDS5ReplicatedAttributeListTotal:[[:blank:]]*\(.*\)$/\1/p" \
+        <<< "$cmd_output")
+      # Remove krblastsuccessfulauth from the list of excluded total
+      # replication attributes
+      new_repl_attr_total=${old_repl_attr_total// krblastsuccessfulauth/}
+
+      # Update the topology segment so that krblastsuccessfulauth is
+      # removed from the lists of fractional and total replication
+      # exclusions.
+      #
+      # Note that it is harmless to run this command when it changes
+      # nothing; however, we must temporarily turn off the bash option
+      # errexit since in that case the error code indicates a failure.
+      set +o errexit
+      ipa topologysegment-mod "$suffix" "$segment" \
+        --replattrs="$new_repl_attr" --replattrstotal="$new_repl_attr_total"
+      set -o errexit
+    done
+  done
+}
+
+# Re-initialize this replica.  This should be run after the topology
+# segments are altered by the enable_last_successful_auth_replication
+# function so that data is immediately synchronized.
+function reinitialize_replica {
+  # The suffix just needs to be valid, since every topology suffix
+  # with the same name but a different topology suffix should be
+  # otherwise identical.
+  first_suffix="${TOPOLOGY_SUFFIXES[0]}"
+  my_hostname=$(hostnamectl status --static)
+
+  # Find a topology segment between the server where this script is
+  # being run and the server against which we created the replica.
+  # From the name of that segment extract the hostname of the server
+  # against which we created the replica.
+  #
+  # Note that we only print the first segment name that is a match
+  # since any match will work.
+  #
+  # The T in the sed command means "jump if no substitution took
+  # place" which means that the p and q commands will ONLY run if a
+  # substitution took place; otherwise, we proceed to the next line.
+  #
+  # This command may fail if the topology segment we're looking for is
+  # the other way around, i.e., if the server where we're running this
+  # script is the left node in the topology segment.  In that case
+  # segment_name will be empty and we will run a different command
+  set +o errexit
+  segment_name=$(ipa topologysegment-find "$first_suffix" --pkey-only \
+    --rightnode="$my_hostname" \
+    | sed --quiet "s/^[[:blank:]]*Segment name:[[:blank:]]*\(.*\)$/\1/; T; p; q")
+  set +o errexit
```

Comment:
Did you mean to re-enable it?

```suggestion
  set -o errexit
```